src/durableorchestrationclient.ts

```diff
@@ -324,13 +325,8 @@ export class DurableOrchestrationClient implements DurableClient {
         }
     }
 
-    public async raiseEvent(

```

Comment:
I'm ok either way.

Reply:
Not a big fan of this final option with a `OrchestrationEvent` type - that feels like 2 options objects instead of one.
I don't feel too strongly one way or another for the remaining options. @hossam-nasr - what do you think?

Reply:
+1 to your goal statement.

Regarding the scenarios - these are all valid in DF. In general, `raiseEvent` is just a generic "send and forget" API and so all these scenarios could make sense depending on the application, and the event receiver's design.

As all these scenarios are valid, I recommend keeping it simple and opting for your first suggestion:
`raiseEvent(instanceId: string, eventName: string, eventData: unknown, options?: TaskHubOptions): Promise<void>;`

I think that's the least assuming choice: it only bundles the optional parameters into an options object, which I think is the  simplest design.

Reply:
This all makes sense to me. This decision was one of those I was conflicted about, but saving the options object for just the optional arguments makes sense to me. I made the update agreed upon:

```TS
raiseEvent(instanceId: string, eventName: string, eventData: unknown, options?: TaskHubOptions): Promise<void>
```